DOCS-1616: Update readme (#5)
viam-labs/berry-imu: README.md

# BerryIMU Module

This module implements the `rdk:movement_sensor` API in a `berryimu` model.
You can use this module to interface with the BerryIMU v3 [accelerometer, gyroscope](https://ozzmaker.com/wp-content/uploads/2020/08/lsm6dsl-datasheet.pdf), [magnetometer](https://ozzmaker.com/wp-content/uploads/2020/08/lis3mdl.pdf), and [barometric altitude sensor](https://www.mouser.com/pdfdocs/BST-BMP388-DS001-01.pdf) using an I2C connection on ARM64 systems.

## Requirements

- [BerryIMU v3](https://ozzmaker.com/product/berryimu-accelerometer-gyroscope-magnetometer-barometricaltitude-sensor/) movement sensor
- Single board computer running Linux Arm64
    - Enable I2C communication on your single board computer. If you are using a Raspberry Pi, see [Enable communication protocols](https://docs.viam.com/get-started/installation/prepare/rpi-setup/#enable-communication-protocols).

> - Install the [Viam Python SDK](https://python.viam.dev/) on your single board computer:
> ```sh
> pip install viam-sdk
> ```

Reviewer: The code sample tab has a new updated set of commands to install the Python SDK in a virtual environment. https://docs.viam.com/build/program/#requirements is being updated in a current PR, so maybe just link to it instead of including detailed instructions?
```suggestion
- [Install the Viam Python SDK](https://docs.viam.com/build/program/#requirements) on your single board computer.
```

Author: Agreed, this keeps the information current. Thank you!

> - Calibrate your IMU sensor by running the following command in the BerryIMU module's `src` directory: 

Reviewer: [opt] Very optional: Consider dividing the "Requirements" section: Delete the "Requirements" heading and replace it with "Hardware requirements," then create a "Prerequisites" or "Setup" or similar heading for the i2c setup, python install, and calibration.

Author: I think this works - what do you think about keeping the Requirements heading and having **Hardware** and **Software**/**Software setup** subheadings? It could be nice to maintain similarity in structure to other readmes. I changed it to include these headings, but very open to feedback changing it more!

Reviewer: I think it's good enough as it currently is!

```sh
python3 calibration.py
```

> [!TIP]
> During calibration, rotate the BerryIMU 360 degrees in all directions (X,Y, and Z) until the script returns calibration values.
> Note the returned calibration values, as you'll need to set these values into your movement sensor component's attributes during configuration. 

## Build and Run 

> To use this module, follow the instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry) and select the `viam-labs:movement_sensor:berryimu` model from the [`berryimu` module]((https://app.viam.com/module/viam-labs/berryimu).

Reviewer: ```suggestion
To use this module, follow the instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry) and select the `viam-labs:movement_sensor:berryimu` model from the [`berryimu` module](https://app.viam.com/module/viam-labs/berryimu).
```

Author: good catch thank you!

| `hard_iron_x_max` |  int   | **Required** | Maximum value for the x-axis hard iron distortion correction.  |
| `soft_iron_x_min` |  int   | **Required** | Minimum value for the x-axis soft iron distortion correction.  |
| `soft_iron_y_max` |  int   | **Required** | Maximum value for the y-axis soft iron distortion correction.  |
| `soft_iron_y_min` |  int   | **Required** | Minimum value for the y-axis soft iron distortion correction.  |
| `hard_iron_y_min` |  int   | **Required** | Minimum value for the y-axis hard iron distortion correction.  |
| `calibrate` |  bool   | Optional | Flag indicating whether callibration is enabled to alibrate the magnetometer using the min and max values (true is enabled). |
| `i2c_bus` |  str   | **Required** | Name of the I2C bus wired to the IMU. |

#### Example configurations

```json
{
    "hard_iron_y_max": 1953,
    "soft_iron_x_max": 1069,
    "hard_iron_x_min": -1953,
    "hard_iron_x_max": 118,
    "soft_iron_x_min": -1069,
    "soft_iron_y_max": 1069,
    "soft_iron_y_min": -1056,
    "hard_iron_y_min": -1029,
    "calibrate": true,
    "i2c_bus": "1"
}
```
```json
{
    "calibrate": false,
    "i2c_bus": "1"
}
```

> ###  Wire your Berry IMU movement sensor

Reviewer: ```suggestion

###  Wire your Berry IMU movement sensor
```
[nit] I think we usually add newlines around code blocks

Wire the board pins outlined below to the corresponding pins on your IMU sensor. The IMU can be powered with the 3.3v pin, and also supports 5V.

| Board Pin  | Berry IMU Pin |
| ------------- | ------------- |
|      SCL      |      SCL      |
|      SDA      |      SDA      |
|      3.3V     |     3.3V      |

Verify the I2C connection by running the following command on your board:

```sh
    i2cdetect -y 1
```

If the IMU is wired correctly, the output of the command will be the I2C addresses `1c` and `6a`. This indicates that the IMU is successfully connected and detected by your system.

### Next Steps

- To test your movement sensor, go to the [**Control tab**](https://docs.viam.com/manage/fleet/robots/#control).
- To write code against your movement sensor, use one of the [available SDKs](https://docs.viam.com/program/).

## Troubleshooting

> To troubleshoot if you encounter an `OSError: 5`, check your I2C connection. Ensure your wiring is correct and that the `i2cdetect` command successfully detects the device's I2C addresses.

Reviewer: ```suggestion
If you encounter an `OSError: 5`, check your I2C connection. Ensure your wiring is correct and that the `i2cdetect` command successfully detects the device's I2C addresses.
```

# BerryIMU Module

This module implements the `rdk:movement_sensor` API in the `berryimu` model.

> You can use this module to interface with the BerryIMU v3 [accelerometer, gyroscope](https://ozzmaker.com/wp-content/uploads/2020/08/lsm6dsl-datasheet.pdf), [magnetometer](https://ozzmaker.com/wp-content/uploads/2020/08/lis3mdl.pdf), and [barometric altitude sensor](https://www.mouser.com/pdfdocs/BST-BMP388-DS001-01.pdf) using an I2C connection on ARM64 systems.

Reviewer 2: We don't currently support the barometric sensor.

| `soft_iron_y_min` |  int   | **Required** | Minimum value for the y-axis soft iron distortion correction.  |
| `hard_iron_y_min` |  int   | **Required** | Minimum value for the y-axis hard iron distortion correction.  |
| `calibrate` |  bool   | Optional | Flag indicating whether callibration is enabled to alibrate the magnetometer using the min and max values (true is enabled). |
| `i2c_bus` |  str   | **Required** | Name of the I2C bus wired to the IMU. |

#### Example configurations

```json
{
    "hard_iron_y_max": 1953,
    "soft_iron_x_max": 1069,
    "hard_iron_x_min": -1953,
    "hard_iron_x_max": 118,
    "soft_iron_x_min": -1069,
    "soft_iron_y_max": 1069,
    "soft_iron_y_min": -1056,
    "hard_iron_y_min": -1029,
    "calibrate": true,
    "i2c_bus": "1"
}
```

```json
{
    "calibrate": false,
    "i2c_bus": "1"
}
```
###  Wire your Berry IMU movement sensor

Wire the board pins outlined below to the corresponding pins on your IMU sensor. The IMU can be powered with the 3.3v pin, and also supports 5V.

| Board Pin  | Berry IMU Pin |
| ------------- | ------------- |
|      SCL      |      SCL      |
|      SDA      |      SDA      |

> |      3.3V     |     3.3V      |

Reviewer 2: should add board gnd -> imu gnd

# BerryIMU Module

This module implements the `rdk:movement_sensor` API in the `berryimu` model.
You can use this module to interface with the BerryIMU v3 [accelerometer, gyroscope](https://ozzmaker.com/wp-content/uploads/2020/08/lsm6dsl-datasheet.pdf) and [magnetometer](https://ozzmaker.com/wp-content/uploads/2020/08/lis3mdl.pdf) using an I2C connection on ARM64 systems.

> ## Requirements

Reviewer: ```suggestion
## Requirements

```
[nit] Add newlines around headings
